Source/FeatureRegistry.swift

```diff
@@ -132,6 +132,39 @@ extension LabeledGroupItem { /* Collection Support */
             }
         }
     }
+
+
+}
+
+internal extension Collection where Element == LabeledItem {
+
+
+    /// This function recursivly traverses through the elements in it's array and returns an array of transformed objects.
+    ///
+    /// - Parameters:
+    ///   - groupItems: Array of `LabelGroupItems` types that keeps track of the parent group items of a feature.
+    ///   - resultBuilder: Configuration block for creating the return type at the end of the tree
+    ///   - filter: Includes the element if the given predicate is satisfied
+    func depthFirstMap<T>(groupItems: [LabeledGroupItem] = [],
```

Comment:
I had a bit if difficulty with the naming of this function, suggestions are welcome!

Reply:
I think that `depthFirstCompactMap` is slightly closer since it does filter?  It is `internal` API so an overly descriptive name is acceptable.

Reply:
updated